libs/state/selections/src/lib/accumulation-observable.ts

```diff
@@ -20,16 +20,36 @@ import {
   withLatestFrom,
 } from 'rxjs/operators';
 import { AccumulationFn, Accumulator } from './model';
+import { inject, InjectionToken } from '@angular/core';
 
 const defaultAccumulator: AccumulationFn = <T>(st: T, sl: Partial<T>): T => {
   return { ...st, ...sl };
 };
 
+/**
+ * Injection token for the default accumulator function.
+ *
+ * @example
+ * providers: [
+ *  {
+ *   provide: RX_ACCUMULATOR_FN,
+ *   useValue: (state, slice) => ({ ...state, ...slice })
+ *  }
+ * ]
+ */
+export const RX_ACCUMULATOR_FN = new InjectionToken<AccumulationFn>(
+  'RX_ACCUMULATOR_FN',
+  {
+    factory: () => defaultAccumulator,
+  }
+);
+
 export function createAccumulationObservable<T extends object>(
   stateObservables = new Subject<Observable<Partial<T>>>(),
-  stateSlices = new Subject<Partial<T>>(),
-  accumulatorObservable = new BehaviorSubject(defaultAccumulator)
+  stateSlices = new Subject<Partial<T>>()
 ): Accumulator<T> {
+  const accumulatorFn = inject(RX_ACCUMULATOR_FN);
```

Comment:
should this be `{ optional: true, self: true }` ?

We probably don't want accumulators to trickle down. Imo this is very specific behavior per instance.

Reply:
If we mark self: as true then we can't specify the token for multiple components using the DI three. Not sure it is what we want. It also has no need to be optional as there's a default factory, we want a default value anyway so the token should not be optional.

Reply:
I don't understand how the defaultfactory works. Imo `inject` throws an error if it cannot find a provider for a requested token.

Anyway, I think it will be VERY uncommon to have it shared between multiple state instances. Having it without `{ self: true }` would also mean any parent component dictates the accumulator for any given child component. Given the fact the accumulator can change the entire behavior of your state, this is probably not what you want. There is also no chance of controlling this trickle-down effect, as it's baked into the DI tree. 

I'm still for having it `{ self: true }`. In case someone wants to have multiple states with the same accumulator, we can help how to define a wrapper function

Reply:
let's ask for more opinions on this. @Karnaukhov-kh @eneajaho @BioPhoton